src/OnChain.hs

```diff
@@ -117,6 +122,7 @@ validateUnregister ScriptParams{..} EnRegistration{..} ctx
     |   noScriptOutputs $ txInfoOutputs info
       -- the ENOPNFT must be burnt in this transaction
       , isEnOPNftBurnt
+      -- TODO You mean EnOpNFT instead of ENNFT?
```

Comment:
The ENOP-NFT must be burnt in unregister transaction, so the EN-NFT can be unlocked and sent back.